Modified documentation

spolifroni-amd opened the pull request:
Changes include:

* Updating the landing page
* Removing Search and Index links
* Updating the What Is using text from "Inference optimization with MIGraphX"
* Creating an install and build section
* Changing the title of "contributing to MIGraphX"
* Changing the intro of "contributing" to smooth it out
* Updated the index file
* Updated the toc

causten reviewed (changes requested):
I added some comments but if you are going to spend time modifying the documentation it would be good to make a clear distinction between the developer and user use of the product.  Separate out the installing and building.  A developer is interested in building the code, a user is interested in installing the code.  Mixing them together creates confusion.

There is only one way to build the code since all three use rbuild.  Having two other ways is pointless.

CharlieL7 reviewed (approved):
Good starting point for updating the documentation. I agree with the changes Chris proposed. Main change would be simplifying the installation section to just use `rbuild`. Hopefully we find some time later to add in a section to teach how to use MIGraphX if you just want to run models. Approving, but please make the changes Chris mentioned.

spolifroni-amd commented on docs/install/prerequisites.rst:
```diff
@@ -0,0 +1,22 @@
+.. meta::
```
OK noted.

spolifroni-amd commented:
> I added some comments but if you are going to spend time modifying the documentation it would be good to make a clear distinction between the developer and user use of the product. Separate out the installing and building. A developer is interested in building the code, a user is interested in installing the code. Mixing them together creates confusion.

This was a first pass at exactly that. This is why "build" and "install" are separate sections. I'll make it clear at the top of the build sections that this is for developing for and contributing to MIGraphX rather than for using the MIGraphX API.
> 
> There is only one way to build the code since all three use rbuild. Having two other ways is pointless.

Which steps from which doc would you prefer me to keep?

spolifroni-amd commented on docs/index.rst:
```diff
@@ -8,31 +8,37 @@
+
+    * :doc:`Prerequisites <./install/prerequisites>`
+    * :doc:`Install MIGraphX <./install/install>`
+    * :doc:`Install MIGraphX with the package installer <./install/installing_with_package>`
```
So should I keep the instructions as written out in rbuild or cmake. Because they're a little different.

spolifroni-amd commented:
> Good starting point for updating the documentation. I agree with the changes Chris proposed. Main change would be simplifying the installation section to just use `rbuild`. Hopefully we find some time later to add in a section to teach how to use MIGraphX if you just want to run models. Approving, but please make the changes Chris mentioned.

I really do want to add that section too. Hopefully soon.

spolifroni-amd commented:
I've removed all the build sections except build with rbuild, removed the prerequisites, and moved What is to the front of the page. I also added a link to the github repo at the top.

causten commented on docs/what-is-migraphx.rst:
```diff
@@ -8,6 +8,4 @@
-AMD MIGraphX is AMD's graph inference engine that accelerates machine learning model inference. The optimized execution engine is useful for deep learning neural networks.
-
-You can utilize MIGraphX functions using :ref:`C++ APIs <cpp-api-reference>`, :ref:`Python APIs <python-api-reference>`, and the :ref:`command-line tool migraphx-driver <migraphx-driver>`.
+AMD 
```
I ask because it seems to already be covered in docs/index.rst.  Along with it being a only 1 acronym long sentence

spolifroni-amd replied:
I was supposed to delete that and the file. Thanks for catching that! Will fix it up post haste!
